training/unicharset_extractor.cpp

```diff
@@ -50,7 +50,9 @@ static void AddStringsToUnicharset(const GenericVector<STRING>& strings,
                                      /*report_errors*/ true,
                                      strings[i].string(), &normalized)) {
       for (const string& normed : normalized) {
-        if (normed.empty() || IsWhitespace(normed[0])) continue;
+
+       // normed[0] has converted to utf-8
+        if (normed.empty() || IsUTF8Whitespace(normed.c_str())) continue;
```

Comment:
In theory that new code should work (although the comment is not quite correct).

The code fails nevertheless because IsUTF8Whitespace() does not work as expected: when a UTF-8 character like "«" is seen, the function calls SpanUTF8Whitespace() which calls IsWhitespace() which should get the 32 bit codepoint 0xc2ab, but it only gets 0x00ab. @theraysmith, this looks like a problem in the UNICHAR class.

Reply:
I was wrong. The codepoint for "«" is 0xab. The new code works perfectly.

Reply:
@ivanzz1001, could you fix the comment, please (something like "// normed is a UTF-8 encoded string") and maybe squash your commits? Then this PR is ready to get merged.